### DISCUSSION TITLE
Zigbee2mqtt e openhab

### DISCUSSION THREAD

**Discussion by dany-69** [indomus/forum#401] [HUB: openHAB]
Una domanda...
Ho configurato dei dispositivi con zigbee2mqtt, su browser e mqtt Eclipse li vedo ma non son come abbinarli in openhab?
Ho cercato anche nella comunity di oh ma nada.
Avreste una guida o un file da poterlo studiare?
Grazie

**Comment by KC-inDomus** [comment]
no, purtroppo niente guida lato inDomus.
cmq sia dovresti poter configurare i vari elementi partendo dai topic telemetrici e di comando disponibili sul sito di zigbee2mqtt.

**Reply by dany-69** [reply]
Ho guardato ma non ho capito molto

**Reply by snakuzzo** [reply]
Ti posso consigliare innanzi tutto di studiare e capire bene questa parte -> [Openhab Concepts](https://www.openhab.org/docs/concepts/).
Solo quando ti è ben chiaro tutto quanto puoi procedere con il capire come configurare manualmente gli item (entità) basati sulle things (device) MQTT -> [Things Advanced](https://www.openhab.org/docs/tutorial/things_advanced.html)

Per quanto riguarda il binding guarda anche -> [MQTT Binding](https://www.openhab.org/addons/bindings/mqtt/) che è fondamentale per la definizione di tutto quello che ti occorre.

Ovviamente non essendoci autodiscovery Zigbee2mqtt->Openhab (che io sappia), ti conviene cercare il device che hai configurato tra i [Supported Devices](https://www.zigbee2mqtt.io/information/supported_devices.html) e guardare come sono definite le entità manuali per Home Assistant. Da quello riesci a capire come configurare le "Things" di openhab

**Reply by dany-69** [reply]
Grazie lo farò

**Reply by KC-inDomus** [reply]
domanda: ma stai partendo da zero, su OpenHAB?

**Comment by dany-69** [comment]
Diciamo di sì

Il lun 7 giu 2021, 14:08 KC-inDomus ***@***.***> ha scritto:

**Reply by snakuzzo** [reply]
se sei a digiuno, ti consiglio di utilizzare Home Assistant

**Reply by dany-69** [reply]
Non proprio, ho già creato diverse rules, diversi items e things, il problema è che non riesco ad inserire i dispositivi zigbee2mqtt in Oh. Non so la procedura, con il binding mqtt non riesco

**Reply by snakuzzo** [reply]
guarda, io ieri ci ho messo mano per la prima volta e sono riuscito ad integrare a mano un interruttore MQTT. Per "a mano" intendo creando i file .things e .items...non da webgui.
Devi sapere la struttura dei topic che è indicata nella pagina del device sul progetto zigbee2mqtt (vedi i link che ti ho mandato ieri)

Posso farti vedere un esempio al volo (magari non è completissimo, ma funziona)...

**things/lights.things**
```java
Thing mqtt:topic:swLuceCamera "Luce Camera" (mqtt:broker:mosquitto) {
        Channels:
                Type switch : switch "Camera da letto" [
                        stateTopic="stat/sonoff-camera/RESULT",
                        commandTopic="cmnd/sonoff-camera/POWER",
                        on="ON",
                        off="OFF"
                ]
                Type string : reachable "Reachable" [
                        stateTopic = "tele/sonoff-camera/LWT"
                ]
```


**items/lights.items**
```java
Switch sLuceCamera "Luce Camera"  { channel="mqtt:topic:swLuceCamera:switch"
```

...ovviamente modifica i topic, nomi, etc in base a quello che hai tu

**Reply by dockydock** [reply]
grazie mille, stasera provo, li creo con visual studio code?

**Reply by snakuzzo** [reply]
come ti pare...l' IDE/editor che utilizzerai è soggettivo. I file li puoi creare pure a manina da riga di comando, non serve un editor particolare.
Ovviamente do per scontato che tu abbia creato il binding MQTT

**Reply by dockydock** [reply]
ok, visto che sto usando vsc, uso lui

**Reply by dockydock** [reply]
si, ho anche attivato il mosquito in oh

**Reply by dockydock** [reply]
ho provato così ma non funziona
**THINGS**
'''
Bridge mqtt:broker:f38fca98f5 [ host="localhost", port=1883, secure=false, username="openhabian", password="openhabian", clientID="openHAB" ]
{
//Contatti Magnetici
    Thing topic Contatto_Magnetico_Ingresso "Contatto_Magnetico_Ingresso" {
    Channels:
    Type number : Battery  [ stateTopic="zigbee2mqtt/Contatto_Magnetico_Ingresso", transformationPattern="JSONPATH:$.Battery" ]
	Type switch : contact  [ stateTopic="zigbee2mqtt/Contatto_Magnetico_Ingresso", transformationPattern="JSONPATH:$.Contact", on="true", off="false" ]
	Type number : Linkquality  [ stateTopic="zigbee2mqtt/Contatto_Magnetico_Ingresso", transformationPattern="JSONPATH:$.Linkquality" ]
    Type number : Battery_low  [ stateTopic="zigbee2mqtt/Contatto_Magnetico_Ingresso", transformationPattern="JSONPATH:$.Battery_low" ]
      }
'''
**ITEMS**
'''
//Contatti Magnetici
Number Contatto_Magnetico_Ingresso_Battery  "Contatto Magnetico Ingresso Batteria [%d %%]" <battery> {channel="mqtt:topic:f38fca98f5:Contatto_Magnetico_Ingresso:Battery" } 
Switch Contatto_Magnetico_Ingresso_Contact  "Contatto magnetico Ingresso Contact" <switch>  ["OpenState","Opening"] {channel="mqtt:topic:f38fca98f5:Contatto_Magnetico_Ingresso:Contact" } 
Number Contatto_Magnetico_Ingresso_Linkquality  "Contatto Magnetico Ingresso Link Quality [%.1f lqi]" <signal> {channel="mqtt:topic:f38fca98f5:Contatto_Magnetico_Ingresso:Linkquality" } 
Number Contatto_Magnetico_Ingresso_Battery_Low  "Contatto Magnetico Ingresso Battery Low"  <signal>  {channel="mqtt:topic:f38fca98f5:Contatto_Magnetico_Ingresso:Battery_Low" } 
'''
sicuramente c'è qualcosa di sbagliaot

**Reply by KC-inDomus** [reply]
(butta un occhio a Home Assistant.)

**Reply by dockydock** [reply]
anche se uso openhab?

**Reply by dockydock** [reply]
comunque:

'''
THINGS

contattomagneticoingresso.things

Thing mqtt:topic:Contatto_Magnetico_Ingresso "Contatto_Magnetico_Ingresso" (mqtt:broker:mosquitto) {
        Channels:
            Type switch : switch "Contatto Magnetico Ingresso" [
                          stateTopic="stat/Contatto_Magnetico_Ingresso/RESULT",
                          commandTopic="cmnd/Contatto_Magnetico_Ingresso/POWER",
                          on="true",
                          off="false"
}
               
ITEMS

contattomagneticoingresso.items

Switch Contatto_Magnetico_Ingresso_Contact "Contatto magnetico Ingresso Contact" <switch>  { channel="mqtt:topic:Contatto_Magnetico_Ingresso:Contact" }

'''
ho modificato il tuo con i miei dati, non so se funziona perchè sono al lavoro (stasera provo), per il canale batteria, batteria low e linkquality?

**Reply by dockydock** [reply]
> (butta un occhio a Home Assistant.)

il link per vedere?

**Reply by dockydock** [reply]
> guarda, io ieri ci ho messo mano per la prima volta e sono riuscito ad integrare a mano un interruttore MQTT. Per "a mano" intendo creando i file .things e .items...non da webgui.
> Devi sapere la struttura dei topic che è indicata nella pagina del device sul progetto zigbee2mqtt (vedi i link che ti ho mandato ieri)
> 
> Posso farti vedere un esempio al volo (magari non è completissimo, ma funziona)...
> 
> **things/lights.things**
> 
> ```java
> Thing mqtt:topic:swLuceCamera "Luce Camera" (mqtt:broker:mosquitto) {
>         Channels:
>                 Type switch : switch "Camera da letto" [
>                         stateTopic="stat/sonoff-camera/RESULT",
>                         commandTopic="cmnd/sonoff-camera/POWER",
>                         on="ON",
>                         off="OFF"
>                 ]
>                 Type string : reachable "Reachable" [
>                         stateTopic = "tele/sonoff-camera/LWT"
>                 ]
> ```
> 
> **items/lights.items**
> 
> ```java
> Switch sLuceCamera "Luce Camera"  { channel="mqtt:topic:swLuceCamera:switch"
> ```
> 
> ...ovviamente modifica i topic, nomi, etc in base a quello che hai tu

come ho modificato io (vedi codice) non funziona
